async_producer.go

```diff
@@ -555,6 +555,21 @@ func (tp *topicProducer) partitionMessage(msg *ProducerMessage) error {
 		return err
 	}
 
+	// Select only partitions with leaders in this rack if configured so, falling back if none are available.
+	if tp.parent.conf.Producer.PartitionerRackAware {
+		var availablePartitionsInRack []int32
+		var clientRack = tp.parent.client.Config().RackID
```

Comment:
When making a declaration and assignment it’s recommended to use the short-hand notation:
```suggestion
		clientRack := tp.parent.client.Config().RackID
```